### ISSUE TITLE
Refactoring breaks formatting

### ISSUE THREAD

**Issue by NickSdot** [BashSupport-Pro/bashsupport-pro#109]
I am facing some problems using the refactoring feature, which you [confirmed](https://github.com/BashSupport-Pro/bashsupport-pro/issues/108#issuecomment-1133414456) comes from BSP. Using latest IntelliJ  2022.1.1, with latest BSP.

## Example 1

I used the following method:

1. right click on `command.init2.zsh`
2. click refactor
3. click rename
4. give new name

**Before:**
```bash
 cd "$(dirname "${BASH_SOURCE[0]}")" \
    && . "../../../scripts/command.init.zsh" \
    && . "../../../scripts/init.zsh" \
    && . "dummy.sh"
```

**After:**
```bash
 cd "$(dirname "${BASH_SOURCE[0]}")" \
    && ."../../command.init2.zsh"&& . "../../../scripts/init.zsh" \.
    && . "dummy.sh"
```

**Problems**
- Beginning of the first sourcing line `&& ."` is missing a space before `"`
- Sourcing of `command.init2.zsh"` is missing the space and `\` at the end of the line
- `scripts/init.zsh` is not on a new line

<img width="1498" alt="image" src="https://user-images.githubusercontent.com/32384907/169626779-9d739d2b-0eac-4a54-b826-94b430da24b4.png">

## Example 2

This is honestly a bit made up example. I can't remember anymore in which exact part of my code this happened before and I am using a non existing path to be able to reproduce the issue. However, it should make clear what the problem is.

I used the following method:

1. right click on `file1.sh`
2. click refactor
3. click move
4. give new path

**Before:**
```bash
  cd "$(dirname "${BASH_SOURCE[0]}")" \
    && . "../file1.sh" \
    && . "../file2.sh" \
    && . "../file3.sh"

  cd "$(dirname "${BASH_SOURCE[0]}")" \
    && . "../scripts/command.init.zsh" \
    && . "../scripts/init.zsh" \
    && . "../scripts/core.config.sh" \
    && . "../scripts/helpers/formatter.init.zsh" \
    && . "../scripts/helpers/content.helpers.sh" \
    && . "../scripts/helpers/utils.sh"

  message \
    --level="error" \
    --style="list" \
    --title="Title" \
    --message="Message Message Message Message Message Message Message
               Message Message Message Message Message" \
    --no-icons
```

**After:**
```bash
  cd "$(dirname "${BASH_SOURCE[0]}")" \
    && . "../file1.sh" \
    && . "../file2.sh" \
    && . "../file3.sh"

  cd "$(dirname "${BASH_SOURCE[0]}")" \
  && . "../scripts/command.init.zsh" \
  && . "../scripts/init.zsh" \
  && . "../scripts/core.config.sh" \
  && . "../scripts/helpers/formatter.init.zsh" \
  && . "../scripts/helpers/content.helpers.sh" \
  && . "../scripts/helpers/utils.sh"  message \
    --level="error" \
    --style="list" \
    --title="Title" \
    --message="Message Message Message Message Message Message Message
               Message Message Message Message Message" \
    --no-icons
```

**Problems**
- Line 288 has no new path applied at all.
- Non wanted changes from line 293 to 299; refactoring added one new level to the file oath
- `message` function call isn't any longer where it belongs to, and hence broken.

<img width="1583" alt="image" src="https://user-images.githubusercontent.com/32384907/169627385-1cf0b0a7-1411-4746-95c6-aca184d07e67.png">

---

Thanks for having a look!

**Comment by jansorg** [comment]
Thanks for reporting! 

I wasn't able to reproduce the incorrect insertions of `../` into the file, but all of the other problems stem from a problem with partial reformatting, which is called after refactorings. BashSupport Pro attempts to provide support for parts of a file, even though the underlying shfmt isn't offering this. This doesn't seem to work too well here, though. In the long run it will have to provide its own formatter...

I'll release a fix for 2.1.x shortly, which turns off the formatting during refactorings. A rename should only change the modified parts at most and nothing else.

**Comment by NickSdot** [comment]
Hi @jansorg. Quick confirmation from my side that the he latest updated fixed the issue. Thanks!
